File: frontend/quiz/src/seleniumTests/Pages/MainPage.js
import Page from "./Page";
import { getElementById } from "./Utility";
class MainPage extends Page {
  async isVisible() {
    return super.isVisibleById("container-main");
  }
  get testBtn() {
    return getElementById(this.driver, "TestBtn");
  }
  get pythonBtn() {
    return getElementById(this.driver, "PythonBtn");
  }
  async open() {
    await super.open("main");
  }
}

export default MainPage;

File: backend/selenium/MainPage.py
from Utility import getElementById, isVisibleById
from Page import Page
TestBtnId = "TestBtn"
PythonBtnId = "PythonBtn"
IdForVisibility = "container-main"


class MainPage(Page):

    def isVisible(self):
        return isVisibleById(self.driver, IdForVisibility)

    def testBtn(self):
        return getElementById(self.driver, TestBtnId)

    def pythonBtn(self):
        return getElementById(self.driver, PythonBtnId)

    def open(self):
        super.open("main")
